hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientSmartInvocationServiceImpl.java

```diff
@@ -80,6 +90,15 @@ private Connection getConnection(Address target) throws IOException {
         return connection;
     }
 
+    private Connection getOrConnect(Address target) throws IOException {
```

Comment:
Yes, triggerConnect does not wait for connection establishment but just triggers a connect but this method waits for the connection establishment.